source/base/parameter_handler.cc

```diff
@@ -1409,7 +1408,41 @@ bool ParameterHandler::read_input_from_string (const char *s)
 {
   // create an istringstream representation and pass it off
   // to the other functions doing this work
-  std::istringstream in (s);
+
+  // concatenate the lines ending with "\"
+
+  std::ifstream file(s);
+  std::string str;
+  std::string file_contents;
+  bool is_concatenated(false);
+  while (std::getline(file, str))
+    {
+      // if this str must be concatenated with the previous one
+      // we strip the whitespaces
+      if (is_concatenated)
+        {
+          const std::size_t str_begin = str.find_first_not_of(" \t");
+          if (str_begin == std::string::npos)
+            str = ""; // no content
+          else
+            str.erase(0,str_begin); // trim whitespaces from left
+        }
+
+      if (std::find(str.begin(),str.end(),'\\') == str.end())
+        {
+          is_concatenated = false; // next line must be concatenated = false
+          file_contents += str;
+          file_contents.push_back('\n');
+        }
+      else
+        {
+          str.erase(str.find_last_not_of(" \t") + 1); // trim withespaces after "\"
+          str.erase(str.find_last_of('\\')); // remove "\" from the end of line
```

Comment:
You somehow need to test that the backslash really is the last character on the line. Right now, you're simply deleting the last occurrence, but that might be somewhere in the middle where it is desired.

I think you need to check in the `if` condition on line 1431 that indeed the backslash is the last non-space character on the line, not whether there *is* or *is not* a backslash on the line.